client/src/components/CountrySelector/index.js

```diff
@@ -0,0 +1,170 @@
+    this.setState({ currentCountry: country });
+  }
+
+  handleChangeCountry = (e) => {
+    const query = e.target.value.toLowerCase().trim();
+
+    const displayedCountries = COUNTRIES
+      .filter(item => item.countryName.toLowerCase().search(query) >= 0);
+
+    this.setState({ displayedCountries });
+  }
+
+  focusInput = (component) => {
+    if (component) {
+      component.focus();
+    }
+  }
+
+  show = (e) => {
+    this.setState({ isListVisible: true });
+
+    if (typeof window !== 'undefined' && e.target.tagName.toUpperCase() !== 'INPUT') {
+      document.addEventListener('click', this.hide);
+    }
+  }
+
+  hide = (e) => {
+    if (typeof window !== 'undefined' && e.target.tagName.toUpperCase() !== 'INPUT') {
+      this.setState({ isListVisible: false });
+      document.removeEventListener('click', this.hide);
+    }
+  }
+
+  render() {
+    const {
+      currentCountry,
+      isListVisible,
+      displayedCountries,
+    } = this.state;
+
+    return (
+      <div className="country-selector-container">
+        <div className="dropdown">
+          <div className="btn dropdown-toggle c-select" onClick={this.show}>
+            <span>
+              <div
+                alt="flag"
+                className={
+                  currentCountry.ISOALPHA2Code
+                      ? `flag ${currentCountry.ISOALPHA2Code.toLowerCase()} fnone c-dropdown-flag`
+                      : 'c-dropdown-flag'
+                }
+              />{currentCountry.countryName}
+            </span>
+          </div>
+          <div
+            className="dropdown-menu c-dropdown-menu"
+            style={isListVisible ? { display: 'block' } : { display: 'none' }}
+          >
+            <input
+              className="c-dropdown-input"
+              type="text"
+              ref={component => this.focusInput(component)}
+              onInput={this.handleChangeCountry}
+            />
+            <div className="c-dropdown-menu-overflow">
+              <ul className="c-dropdown-ul">{
+                displayedCountries.map((item, index) => {
+                    const flag = item.ISOALPHA2Code.toLowerCase();
+                    return (
+                      <li
+                        tabIndex={index}
+                        className="dropdown-item c-dropdown-item"
+                        data-country={item.ISOALPHA2Code}
+                        key={item.ISOALPHA2Code}
+                        onClick={() => this.handleSelectCountry(item)}
+                      >
+                        <div data-option={`${item.ISOALPHA2Code}`}>
+                          <div className={`flag ${flag} fnone c-dropdown-flag`} /> {item.countryName}
+                        </div>
+                      </li>
+                    );
+                })
+                }
+              </ul>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+CountrySelector.propTypes = {
```

Comment:
It is not a bug, because you don't use `this.props` in `getCountry` method:
Move **defaultProps** and **propTypes** to the class:
```
class CountrySelector extends React.Component {
  static defaultProps = {...};
  static propTypes = {...}
```

Reply:
Well, the fact that I do not use `this.props` does not mean that I am not using this prop...
And I think that to write `CountrySelector.propTypes = {...}` is just the same as to write it inside the class as you proposed... But I will try your solution, who knows maybe it will work out... :)